frontend/packages/console-app/src/components/cloud-shell/MultiTabbedTerminal.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Tabs, Tab, TabTitleText, TabTitleIcon, Button } from '@patternfly/react-core';
+import { CloseIcon, PlusIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import CloudShellTerminal from './CloudShellTerminal';
+import './MultiTabbedTerminal.scss';
+
+const MAX_TERMINAL_TABS = 8;
+
+interface MultiTabbedTerminalProps {
+  onClose?: () => void;
+}
+
+export const MultiTabbedTerminal: React.FC<MultiTabbedTerminalProps> = ({ onClose }) => {
+  const [terminalTabs, setTerminalTabs] = React.useState<number[]>([1]);
+  const [activeTabKey, setActiveTabKey] = React.useState<number>(1);
+  const { t } = useTranslation();
+
+  const addNewTerminal = () => {
+    if (terminalTabs.length < MAX_TERMINAL_TABS) {
+      const tabs = [...terminalTabs];
+      const newTerminalNumber = terminalTabs[terminalTabs.length - 1] + 1;
+      tabs.push(newTerminalNumber);
+      setTerminalTabs(tabs);
+      setActiveTabKey(newTerminalNumber);
+    }
+  };
+
+  const removeCurrentTerminal = (tabIndex: number) => {
+    const tabs = [...terminalTabs];
+    if (tabs[tabIndex] === activeTabKey) {
+      setActiveTabKey(tabIndex > 0 ? tabs[tabIndex - 1] : tabs[tabs.length - 1]);
+    }
+    tabs.splice(tabIndex, 1);
+    setTerminalTabs(tabs);
+  };
+
+  return (
+    <Tabs activeKey={activeTabKey} isBox>
+      {terminalTabs.map((terminalNumber, tabIndex) => (
+        <Tab
+          className="co-multi-tabbed-terminal__tab"
+          translate="no"
+          eventKey={terminalNumber}
+          key={terminalNumber}
+          title={
+            <div>
+              <TabTitleText onClick={() => setActiveTabKey(terminalNumber)}>
+                {t('console-app~Terminal {{number}}', { number: terminalNumber })}
+              </TabTitleText>
+              <TabTitleIcon>
+                {terminalTabs.length > 1 ? (
+                  <Button
```

Comment:
Do we need `style={{padding: 0}}` to ensure that the tab layout isn't affected by PFs default button padding?

Reply:
padding set to 0